numba/core/types/functions.py

```diff
@@ -628,6 +628,10 @@ def generic(self, args, kws):
     def key(self):
         return self.sig, self.cconv, self.get_pointer
 
+    def unify(self, typingctx, other):
+        if isinstance(other, Function):
+            return self
```

Comment:
Weird I admit.  Other suggestions welcome.  Pointers can be function pointers...but of course they could be anything else as well.  I don't remember the details of where this came up but comment it out and I'm sure it will pop up again. 
 @stuartarchibald